chainer/function_hooks/cupy_memory_profile.py

```diff
@@ -124,21 +126,48 @@ def summary(self):
             record['occurrence'] += 1
         return summary
 
-    def _humanized_size(self, size):
-        """Returns a human redable bytes string."""
-        for unit in ['', 'K', 'M', 'G', 'T', 'P', 'E']:
-            if size < 1024.0:
-                return '%3.2f%sB' % (size, unit)
-            size /= 1024.0
-        return '%.2f%sB' % (size, 'Z')
-
-    def print_report(self, file=sys.stdout):
-        """Prints a summary report of memory profiling in functions."""
+    def _choose_unit(self, size):
+        """Choose optimal unit."""
+        denomi = 1
+        for unit in self._units[:-1]:
+            if size / denomi >= 1:
+                return denomi, unit
+            denomi *= 1024.0
+        return denomi, self._units[-1]
+
+    def print_report(self, unit='auto', file=sys.stdout):
+        """Prints a summary report of memory profiling in functions.
+
+        Args:
+            unit (str): Supplementary units used for used memories.
+                `B`, `KB`, `MB`, `GB`, `TB`, `PB`, `EB`, `ZB`, `auto`(default)
+                and `auto_foreach` are supported. If `auto`, units of memories
+                are aligned to the each largest of 'used_bytes' and
```

Comment:
I suppose "the each ~" is not allowed as English.
```suggestion
                are aligned to the largest values of 'used_bytes' and
```

Reply:
Thank you!!